src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.V3.cs

```diff
@@ -61,11 +66,158 @@ public async Task PayloadV3_should_return_all_the_blobs(int blobTxCount)
         Assert.That(getPayloadResultBlobsBundle.Proofs!.Length, Is.EqualTo(blobTxCount));
     }
 
+    [Test]
+    public async Task NewPayloadV3_should_decline_null_args()
+    {
+        MergeTestBlockchain chain = await CreateBlockChain(releaseSpec: Cancun.Instance);
+        IEngineRpcModule rpcModule = CreateEngineModule(chain);
+        ExecutionPayload executionPayload = CreateBlockRequest(
+            CreateParentBlockRequestOnHead(chain.BlockTree), TestItem.AddressD, withdrawals: Array.Empty<Withdrawal>());
+
+        int errorCode = (await rpcModule.engine_newPayloadV3(executionPayload, null!)).ErrorCode;
+
+        Assert.That(errorCode, Is.EqualTo(ErrorCodes.InvalidParams));
+
+        errorCode = (await rpcModule.engine_newPayloadV3(null!, Array.Empty<byte[]>())).ErrorCode;
+
+        Assert.That(errorCode, Is.EqualTo(ErrorCodes.InvalidParams));
+    }
+
+    private const string FurtherValidationStatus = "FurtherValidation";
+
+    [TestCaseSource(nameof(BlobVersionedHashesMatchTestSource))]
+    [TestCaseSource(nameof(BlobVersionedHashesDoNotMatchTestSource))]
+    public async Task<string> NewPayloadV3_should_verify_blob_versioned_hashes_against_transactions_ones
```

Comment:
Bit hard to read, would be good to split into helper methods. Can be local methods.